lib/did_you_mean/spell_checker.rb

```diff
@@ -10,23 +10,24 @@ def initialize(dictionary: )
     end
 
     def correct(input)
-      input     = normalize(input)
-      threshold = input.length > 3 ? 0.834 : 0.77
+      input        = normalize(input)
+      input_length = input.length
```

Comment:
I simply don't think this is  a good practice in this particular case... Sounds too weak to justify when the `#length` method is as fast as local variable lookup.